superagi/helper/github_helper.py

```diff
@@ -335,3 +337,112 @@ def _get_file_contents(self, file_name, agent_id, agent_execution_id, session):
+        url = f'https://api.github.com/repos/{repository_owner}/{repository_name}/pulls/{pull_request_number}/commits'
+        headers = {
+            "Authorization": f"token {self.github_access_token}" if self.github_access_token else None,
+            "Content-Type": "application/json",
+        }
+        response = requests.get(url, headers=headers)
+        if response.status_code == 200:
+            commits = response.json()
+            latest_commit = commits[-1]  # Assuming the last commit is the latest
+            return latest_commit.get('sha')
+        else:
+            logger.warning(f'Failed to fetch commits for pull request: {response.json()["message"]}')
+            return None
+
+
+    def add_line_comment_to_pull_request(self, repository_owner, repository_name, pull_request_number,
+                                         commit_id, file_path, position, comment_body):
+        comments_url = f'https://api.github.com/repos/{repository_owner}/{repository_name}/pulls/{pull_request_number}/comments'
+        headers = {
+            "Authorization": f"token {self.github_access_token}",
+            "Content-Type": "application/json",
+            "Accept": "application/vnd.github.v3+json"
+        }
+        data = {
+            "commit_id": commit_id,
+            "path": file_path,
+            "position": position,
+            "body": comment_body
+        }
+        response = requests.post(comments_url, headers=headers, json=data)
+        if response.status_code == 201:
+            logger.info('Successfully added line comment to pull request.')
+            return response.json()
+        else:
+            logger.warning(f'Failed to add line comment: {response.json()["message"]}')
+            return None
+
+    def get_pull_requests_created_in_last_x_seconds(self, repository_owner, repository_name, x_seconds):
+        """
+        Gets the pull requests created in the last x seconds.
+
+        Args:
+            repository_owner (str): Owner of the repository
+            repository_name (str): Repository name
+            x_seconds (int): The number of seconds in the past to look for PRs
+
+        Returns:
+            list: List of pull request objects that were created in the last x seconds
+        """
+        # Calculate the time x seconds ago
+        time_x_seconds_ago = datetime.utcnow() - timedelta(seconds=x_seconds)
+
+        # Convert to the ISO8601 format GitHub expects, remove milliseconds
+        time_x_seconds_ago_str = time_x_seconds_ago.strftime('%Y-%m-%dT%H:%M:%SZ')
+
+        # Search query
+        query = f'repo:{repository_owner}/{repository_name} type:pr created:>{time_x_seconds_ago_str}'
+
+        url = f'https://api.github.com/search/issues?q={query}'
+        headers = {
+            "Authorization": f"token {self.github_access_token}",
+            "Content-Type": "application/json",
+        }
+
+        response = requests.get(url, headers=headers)
```

Comment:
here we can avoid using url and headers and make method small by passing them as param and avoid url,headers as temp var?